apps/core/scenes/search/DatePicker.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import * as React from 'react';
+import { TripInput } from '@kiwicom/margarita-components';
+import { isSameDay } from 'date-fns';
+import {
+  Icon,
+  RangeDatePicker,
+  type RangeDatePickerProps,
+  type StylePropType,
+  type IconNameType,
+} from '@kiwicom/universal-components';
+
+type Props = {|
+  +style?: StylePropType,
+  onPress: () => void,
+  label: string,
+  icon: IconNameType,
+  value: string,
+  ...$Rest<
+    RangeDatePickerProps,
+    {|
+      +onChangeTempDates: (Array<Date>) => void,
+    |},
+  >,
+|};
+
+type State = {|
+  tempDates: Array<Date>,
+  previousDates: Array<Date>,
+|};
+const parseDatePropsToState = ({ dates }: Props) => {
+  const tempDate = dates ?? [new Date(), new Date()];
+  return {
+    tempDates: tempDate,
+    previousDates: tempDate,
+  };
+};
+class DatePicker extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      tempDates: props.dates,
+      previousDates: props.dates,
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    if (
+      !isSameDay(nextProps.dates[0], prevState.previousDates[0]) ||
+      !isSameDay(nextProps.dates[1], prevState.previousDates[1])
+    ) {
+      return parseDatePropsToState(nextProps);
+    }
+    return null;
+  }
+
+  onChangeTempDates = (dates: Array<Date>) => {
```

Comment:
`$ReadOnlyArray`

Reply:
Are you sure about this one? I'm setting this array to state and I think it should be normal Array

Reply:
There is nothing wrong with having the array as ReadOnly. 
```
type State = {|
  dates: $ReadOnlyArray<Dates>
|}
```

Should be valid. You can set dates in state, but not write to this.state.date[0].
Or does flow complain?

Reply:
OK, fixed